src/utilities/include/utilities/Interval.h

```diff
@@ -43,6 +47,124 @@ class Interval
 	{
 	}
 
+	/// Check if the first interval is smaller than the second.
+	/** Interval I1 is smaller than Interval I2 if
+	 * 1. the lower bound of I1 is smaller than the lower bound of I2,
+	 * 2. they have the same lower bound and the upper bound of I1 is smaller than the upper bound
+	 * of I2.
```

Comment:
The semantics is debatable: Usually, if you infer some constraints on an interval, the semantics is, that the constraint holds for all values inside the interval. With respect to this, [a,b] < [c,d] only if b < c (ignoring different bound types and unbounded intervals).

Reply:
Yes, your semantics makes more sense! That means we need to define `operator==` differently, but that also makes sense. I'll update accordingly.

Reply:
Fixed!